docs/content/leginon/Revert_Settings.md
Revert Settings is a tool for use with the Leginon image acquisition software.

Leginon settings for the applications are saved in the database during the installation. When a user use Leginon the first time, the settings or the Appion/Leginon administrator user will be loaded. The user can change them and Leginon will remember the new values from then on.

In the case that a user incorrectly modifies Leginon application settings, the user or an administrator may revert all the settings of a specific user to the default values.

1.  Click on the Revert Settings icon
2.  Choose the user in the list, and click on the Revert button.

Expected result:

-   After restarting Leginon with the same application, the settings that were changed should be reverted.
-   Only those in administrator group or the user in question can do this. A user with Guest permissions may not revert settings.


docs/content/appion/Appion_config_advanced.md
1.  Enable the processing plug-in by uncommenting out the following line in the file`myamiweb/config.php`
        addplugin("processing");

    
     
2.  IMAGIC and Other features:
        // Check if IMAGIC is installed and running, otherwise hide all functions
        define('HIDE_IMAGIC', false);

        // hide processing tools still under development.
        define('HIDE_FEATURE', true);

    
     
3.  Add processing host information
    **processing host is the computer with pbs server running (i.e. the head node of a computer cluster if applicable)**
     
    **Appion version 2.2 and later:**
    The following code should be added and modified for each processing host available.
        $PROCESSING_HOSTS[] = array(
        'host' => 'LOCAL_CLUSTER_HEADNODE.INSTITUTE.EDU', // for a single computer installation, this can be 'localhost'    
        'nproc' => 32,  // number of processors available on the host, not used
        'nodesdef' => '4', // default number of nodes used by a refinement job
        'nodesmax' => '280', // maximum number of nodes a user may request for a refinement job
        'ppndef' => '32', // default number of processors per node used for a refinement job
        'ppnmax' => '32', // maximum number of processors per node a user may request for a refinement job
        'reconpn' => '16', // recons per node, not used 
        'walltimedef' => '48', // default wall time in hours that a job is allowed to run
        'walltimemax' => '240', // maximum hours in wall time a user may request for a job
        'cputimedef' => '1536', // default cpu time in hours a job is allowed to run (wall time x number of cpu's) 
        'cputimemax' => '10000', // maximum cpu time in hours a user may request for a job
        'memorymax' => '', // the maximum memory a job may use
        'appionbin' => 'bin/', // the path to the myami/appion/bin directory on this host
        'appionlibdir' => 'appion/', // the path to the myami/appion/appionlib directory on this host
        'baseoutdir' => 'appion', // the directory that processing output should be stored in
        'localhelperhost' => '', // a machine that has access to both the web server and the processing host file systems to copy data between the systems
        'dirsep' => '/', // the directory separator used by this host
        'wrapperpath' => '', // advanced option that enables more than one Appion installation on a single machine, contact us for info 
        'loginmethod' => 'USERPASSWORD', // Appion currently supports 'USERPASSWORD'.  'SHAREKEY' is possible, but not preferable. 
        'loginusername' => '', // if this is not set, Appion uses the username provided by the user in the Appion Processing GUI
        'passphrase' => '', // if this is not set, Appion uses the password provided by the user in the Appion Processing GUI
        'publickey' => 'rsa.pub', // set this if using 'SHAREDKEY' (not supported 
        'privatekey' => 'rsa'     // set this if using 'SHAREDKEY'
        );

    
     
    **Appion version 2.1 and prior:**
        // --- Please enter your processing host information associate with -- //
        // --- Maximum number of the processing nodes                                   -- //
        // --- $PROCESSING_HOSTS[] = array('host' => 'host1.school.edu', 'nproc' => 4); -- //
        // --- $PROCESSING_HOSTS[] = array('host' => 'host2.school.edu', 'nproc' => 8); -- //

        // $PROCESSING_HOSTS[] = array('host' => '', 'nproc' => );

    
     
4.  Microscope spherical aberration constant
     
    Not needed for Appion version 2.2 and later. Version 2.1 and earlier only:
        $DEFAULTCS = "2.0";

    
     
5.  Redux server information
     
    This is a very new feature, post Appion 2.2, so you only need this if you are running the trunk or 2.2-redux.
    The Redux server replaces the old mrc_php module and will be released in Appion version 3.0.
    [Information on redux parameters to add to the config.php](/appion/leginonUsing_Redux_to_serve_images_on_myamiweb)


myamiweb/ChangeLog
----------------------
dbem - ChangeLog
----------------------

2009-08-07
Release dbem_1_5_1
		- allow to disable processing plugin

2009-07-22
Release dbem_1_5
		- new image constrast interface
		- image comment, hide, examplar
		- add processing web tool for appion
		- imagic/spider stack viewer
		- require mrc_1_5 php module
		- update to use project_1_2

2009-03-17
Release dbem_1_4_6
		- fix default settings

2008-11-06 
Release dbem_1_4_5
		- fix display in admin
		- update to use with project_1_1 package
		- check if php modules are loaded

2008-03-20 
Release dbem_1_4_4
		- display image comment entered in Leginon
		- xml viewer dataset updated
		- new default database setings in admin page

2008-03-01 
Release dbem_1_4_3
		- tomogram viewer
		- RCT viewer
		- option to connect a "Project database"
		- default database setings in admin page

2006-02-24 
Release dbem_1_4_2
	PHP code for image viewers, XML import/export for applications and calibrations, admin tools for intruments, users.
	Reports generation included.

	Module compatible with apache2
	XML dataset support
	PHP5 compatible!


appion/docker/centos7/config.php
<?php

/**
 *	The Leginon software is Copyright under
 *	Apache License, Version 2.0
 *	For terms of the license agreement
 *	see  http://leginon.org
 *
 */
 
 /**
  *  Please visit http://yourhost/myamiwebfolder/setup
  *  for automatically setup this config file for the
  *  first time.
  */
 

require_once 'inc/config.inc';
define('WEB_ROOT',dirname(__FILE__));

// --- define myamiweb tools base --- //
define('PROJECT_NAME', "myamiweb");
define('PROJECT_TITLE', "Appion in a Docker Container");

// --- define site base path -- //
// --- This should be changed if the myamiweb directory is located -- //
// --- in a sub-directory of the Apache web directory. -- //
// --- ex. myamiweb is in /var/www/html/applications/myamiweb/ then -- //
// --- change "myamiweb to "applications/myamiweb" -- //
define('BASE_PATH',"myamiweb");

define('BASE_URL', "/".BASE_PATH."/");
define('PROJECT_URL', BASE_URL."project/");

// --- myamiweb login --- //
// Browse to the administration tools in myamiweb prior to 
// changing this to true to populate DB tables correctly.
define('ENABLE_LOGIN', false);
define('ENABLE_ANONYMOUS', false);
define('COOKIE_PASSPHRASE', '');

// --- Administrator email title and email address -- //
define('EMAIL_TITLE', 'The name of your institute');
define('ADMIN_EMAIL', "vossman77@yahoo.com");

// --- When 'ENABLE_SMTP set to true, email will send out -- //
// --- via ADMIN_EMIL's SMTP server. --// 
define('ENABLE_SMTP', false);
define('SMTP_HOST', 'mail.institute.edu');	//your smtp host

// --- Check this with your email administrator -- //
// --- Set it to true if your SMTP server requires authentication -- //
define('SMTP_AUTH', false);

// --- If SMTP_AUTH is not required(SMTP_AUTH set to false, -- //
// --- no need to fill in 'SMTP_USERNAME' & SMTP_PASSWORD -- //
define('SMTP_USERNAME', "");
define('SMTP_PASSWORD', "");

// --- Set your MySQL database server parameters -- //
define('DB_HOST', "localhost");		// DB Host name
define('DB_USER', "usr_object");		// DB User name
define('DB_PASS', "Phys-554");		// DB Password
define('DB_LEGINON', "leginondb");	// Leginon database name
define('DB_PROJECT', "projectdb");	// Project database name

// --- default URL for project section --- //
define('VIEWER_URL', BASE_URL."3wviewer.php?expId=");
define('SUMMARY_URL', BASE_URL."summary.php?expId=");
define('UPLOAD_URL', BASE_URL."processing/uploadimage.php");

// --- Set cookie session time -- //
define('COOKIE_TIME', 0);		//0 is never expire. 

// --- defaut user group -- //
define('GP_USER', 'users');

// --- XML test dataset -- //
$XML_DATA = "test/viewerdata.xml";

// --- Set Default table definition -- //
define('DEF_PROCESSING_TABLES_FILE', "defaultprocessingtables.xml");
define('DEF_PROCESSING_PREFIX', "ap");		//update mysql privileges

// --- Set External SQL server here (use for import/export application) -- //
// --- You can add as many as you want, just copy and paste the block -- //
// --- to a new one and update the connection parameters -- //
// --- $SQL_HOSTS['example_host_name']['db_host'] = 'example_host_name'; -- //
// --- $SQL_HOSTS['example_host_name']['db_user'] = 'usr_object'; -- //
// --- $SQL_HOSTS['example_host_name']['db_pass'] = ''; -- //
// --- $SQL_HOSTS['example_host_name']['db'] = 'legniondb'; -- //

$SQL_HOSTS[DB_HOST]['db_host'] = DB_HOST;
$SQL_HOSTS[DB_HOST]['db_user'] = DB_USER;
$SQL_HOSTS[DB_HOST]['db_pass'] = DB_PASS;
$SQL_HOSTS[DB_HOST]['db'] = DB_LEGINON;

// --- path to main --- //
set_include_path(dirname(__FILE__).PATH_SEPARATOR
				.dirname(__FILE__)."/project".PATH_SEPARATOR
				.dirname(__FILE__)."/lib".PATH_SEPARATOR
				.dirname(__FILE__)."/lib/PEAR");

// --- add plugins --- //
// --- uncomment to enable processing web pages -- //
addplugin("processing");


// --- Add as many processing hosts as you like -- //
// --- Please enter your processing host information as in the following example. -- //
// --- if the login and passphrase are false/null	-- //
// --- the username and password that the user   	-- //
// --- logs into appion with will be used.			-- //
$PROCESSING_HOSTS[] = array(
  'host' => 'localhost',
  'nproc' => 1,'nodesdef' => '1','nodesmax' => '1',
  'ppndef' => '1','ppnmax' => '1',
  'reconpn' => '1',
  'walltimedef' => '48','walltimemax' => '240',
  'cputimedef' => '1536','cputimemax' => '10000',
  'memorymax' => '',
  'appionbin' => 'bin/','appionlibdir' => 'appion/',
  'baseoutdir' => 'appion','localhelperhost' => '',
  'dirsep' => '/','wrapperpath' => 'appion',
  'loginmethod' => 'SHAREDKEY','loginusername' => '',
  'passphrase' => '','publickey' => 'rsa.pub',
  'privatekey' => 'rsa',
  'gpus' => array( '0.0' => 'device 0: Tesla', '1' => 'device 1', '2' => 'device 2'), 
  'queues' => array('bigmem','myami')
);


// --- Restrict file server if you want --- //
// --- Add your allowed processing directory as string in the array
$DATA_DIRS = array();

// --- Enable Image Cache --- //
define('ENABLE_CACHE', false);
// --- caching location --- //
// --- please make sure the apache user has read access to this folder --- //
// --- define('CACHE_PATH', "/srv/www/cache/"); --- //
define('CACHE_PATH', "");

// --- define Flash player base url --- //
define('FLASHPLAYER_URL', "/flashplayer/");

// --- define python commands - path --- //

// to download images as TIFF or JPEG
// $pythonpath="/your/site-packages";
// putenv("PYTHONPATH=$pythonpath");

// --- Check if IMAGIC is installed and running, otherwise hide all functions --- //
define('HIDE_IMAGIC', true);

// --- Check if MATLAB is installed and running, otherwise hide all functions --- //
define('HIDE_MATLAB', true);

// --- hide processing tools still under development. --- //
define('HIDE_FEATURE', true);

// --- temporary images upload directory --- //
define('TEMP_IMAGES_DIR', "");

// --- use appion wrapper (prepends APPION_WRAPPER_PATH to all Appion commands) --- //
define('USE_APPION_WRAPPER', false);
// --- define('APPION_WRAPPER_PATH', ""); --- //

// --- default appion base directory to store processing results (Normally `parallel to leginon image directory defined in leginon/leginon.cfg) --- // 
// note:  This should end with the directory named "appion"
define('DEFAULT_APPION_PATH',"");

// --- number of parent directories before the sessionname to append after DEFAULT_APPION_PATH --- // 
// The result is a subdivision of data by user after DEFAULT_APPION_PATH, as configured in each user's leginon/leginon.cfg or upon creation of the Leginon session--- //
define('APPION_PARENT_ADDITION',0);

// --- negative index of the directory before the sessionname used to replace the elemenr of DEFAULT_APPION_PATH --- //
// --- negative index of the directory before the last directory named "leginon"
// in DEFAULT_APPION_PATH used to replace the element of DEFAULT_APPION_PATH
// before the last directory named "appion"--- //
// Thre result is a subdivision of data by user or group before appion/leginon directories.
//define('APPION_PARENT_REPLACEMENT',-2);

// --- sample tracking ---//
define('SAMPLE_TRACK', false);

// --- exclude projects in statistics. give a string with numbers separated by ',' ---//
// --- for example, "1,2" ---//
define('EXCLUDED_PROJECTS',"");

// --- hide testing tools related to specific data sets. --- //
define('HIDE_TEST_TOOLS', true);

// --- A startup message may be shown to the user --- //
define('STARTUP_MESSAGE', "");

// --- Redux server Host and Port information. --- //
// --- define('SERVER_HOST', "redux.schools.edu"); --- //
// --- define('SERVER_PORT', "55123"); --- //
define('SERVER_HOST', "localhost");
define('SERVER_PORT', "55123");

// --- list specifc datasets that have test scripts associated with them --- //
// --- example:
// --- $TEST_SESSIONS = array(
// ---		'zz07jul25b'
// ---		,'zz06apr27c'
// ---		,'zz09feb12b'
// ---		,'zz09apr14b'
// ---		,'zz09feb18c'
// ---	);

$TEST_SESSIONS = array();

?>
